### PR TITLE
Fix container tty resize for docker attach by name

### LINKED ISSUE #1436: Docker API: docker resize fails if short name is used with docker attach
docker create -ps busybox /bin/top
docker start <id>
docker ps .... to look up short name (trusting_saha)
docker attach trusting_saha

Results:  it attaches but does not resize the tty window.

### PR DESCRIPTION
**Pull request by sflxn**
Fix resize issue when using docker attach on shorten name (e.g. frustrating_saha).

Resolves #1436.

### CODE REVIEW

File: lib/apiservers/engine/backends/container.go [hunk 1]
```diff
@@ -858,6 +856,30 @@ func createNewAttachClientWithTimeouts(connectTimeout, responseTimeout, response
 	return plClient, transport
 }
 
+func IsContainerRunning(containerID string) (bool, error) {
```
Comment (fdawg4l) [inline lib/apiservers/engine/backends/container.go]: Doesn't look like you need the bool.  Just return an error and check if err != nil.

File: lib/apiservers/engine/backends/container.go [hunk 2]
```diff
@@ -858,6 +856,30 @@ func createNewAttachClientWithTimeouts(connectTimeout, responseTimeout, response
...
+
+	ci := resp.Payload
+
+	if ci != nil && ci.ContainerConfig != nil {
+		if ci.ContainerConfig.State != nil && strings.ToLower(*ci.ContainerConfig.State) == "running" {
+			return true, nil
+		}
+	}
+
+	return false, nil
```
Comment (fdawg4l) [inline lib/apiservers/engine/backends/container.go]: Re: comment above, return an error in this case.

File: lib/apiservers/engine/backends/container.go (hunk 1 above)
Comment (sflxn) [inline lib/apiservers/engine/backends/container.go]: this was part of another PR.   This branch was based on that branch.

File: lib/apiservers/engine/backends/container.go (hunk 2 above)
Comment (sflxn) [inline lib/apiservers/engine/backends/container.go]: ditto above

**Comment by sflxn** [comment]
just squashed the commits.

File: lib/apiservers/engine/backends/container.go [hunk 3]
```diff
@@ -317,6 +317,11 @@ func (c *Container) ContainerRename(oldName, newName string) error {
 func (c *Container) ContainerResize(name string, height, width int) error {
 	defer trace.End(trace.Begin("ContainerResize"))
 
+	vc := cache.ContainerCache().GetContainer(name)
+	if vc == nil {
+		return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
```
Comment (caglar10ur) [inline lib/apiservers/engine/backends/container.go]: Do not return err here, as you check its existence later via PL. Instead just do
```
vc := cache.ContainerCache().GetContainer(name)
if vc != nil {
name = vic
}
```

File: lib/apiservers/engine/backends/container.go [hunk 4]
```diff
@@ -327,7 +332,7 @@ func (c *Container) ContainerResize(name string, height, width int) error {
 	// Call the port layer to resize
 	plHeight := int32(height)
 	plWidth := int32(width)
-	plResizeParam := interaction.NewContainerResizeParams().WithID(name).WithHeight(plHeight).WithWidth(plWidth)
+	plResizeParam := interaction.NewContainerResizeParams().WithID(vc.ContainerID).WithHeight(plHeight).WithWidth(plWidth)
```
Comment (caglar10ur) [inline lib/apiservers/engine/backends/container.go]: No need to do this if you use the if above

**Comment by caglar10ur** [comment]
Closing this in as I went over all the function on that file @https://github.com/vmware/vic/pull/1434

File: lib/apiservers/engine/backends/container.go (hunk 3 above)
Comment (sflxn) [inline lib/apiservers/engine/backends/container.go]: Good point.

**Pull request closed without merging**